hamocc/cyano.F90

```diff
@@ -61,74 +61,69 @@ SUBROUTINE CYANO(kpie,kpje,kpke,kbnd,pddpo,omask,ptho)
-      REAL :: ttemp,nfixtfac
+  ! Local variables
+  INTEGER :: i,j,k
+  REAL :: oldocetra,dano3
+  REAL :: ttemp,nfixtfac
+
+  intnfix(:,:)=0.0
 
-      intnfix(:,:)=0.0
-      
 !
-! N-fixation by cyano bacteria (followed by remineralisation and nitrification), 
+! N-fixation by cyano bacteria (followed by remineralisation and nitrification),
 ! it is assumed here that this process is limited to the mixed layer
 !
-      DO k=1,kmle
-!$OMP PARALLEL DO PRIVATE(i,oldocetra,dano3,ttemp,nfixtfac) 
-      DO j=1,kpje
-      DO i=1,kpie
+  DO j=1,kpje
+     DO i=1,kpie
         IF(omask(i,j).gt.0.5) THEN
-          IF(ocetra(i,j,k,iano3).LT.(rnit*ocetra(i,j,k,iphosph))) THEN
+           DO k=1,kmle(i,j)
+              IF(ocetra(i,j,k,iano3).LT.(rnit*ocetra(i,j,k,iphosph))) THEN
 
-            oldocetra = ocetra(i,j,k,iano3)
-            ttemp = min(40.,max(-3.,ptho(i,j,k)))
+                 oldocetra = ocetra(i,j,k,iano3)
+                 ttemp = min(40.,max(-3.,ptho(i,j,k)))
 
 ! Temperature dependence of nitrogen fixation, Kriest and Oschlies 2015.
-            nfixtfac = MAX(0.0,tf2*ttemp*ttemp + tf1*ttemp + tf0)/tff
+                 nfixtfac = MAX(0.0,tf2*ttemp*ttemp + tf1*ttemp + tf0)/tff
 
-            ocetra(i,j,k,iano3)=ocetra(i,j,k,iano3)*(1-bluefix*nfixtfac) &
-     &                      +bluefix*nfixtfac*rnit*ocetra(i,j,k,iphosph)
+                 ocetra(i,j,k,iano3)=ocetra(i,j,k,iano3)*(1-bluefix*nfixtfac)  &
+                      &    + bluefix*nfixtfac*rnit*ocetra(i,j,k,iphosph)
 
-            dano3=ocetra(i,j,k,iano3)-oldocetra
+                 dano3=ocetra(i,j,k,iano3)-oldocetra
 
-            ocetra(i,j,k,igasnit)=ocetra(i,j,k,igasnit)-dano3*(1./2.)
+                 ocetra(i,j,k,igasnit)=ocetra(i,j,k,igasnit)-dano3*(1./2.)
 
 ! Note: to fix one mole N2 requires: N2+H2O+y*O2 = 2* HNO3 <-> y=2.5 mole O2.
 ! I.e., to release one mole HNO3 = H+ + NO3- requires 1.25 mole O2
-            ocetra(i,j,k,ioxygen)=ocetra(i,j,k,ioxygen)-dano3*1.25
+                 ocetra(i,j,k,ioxygen)=ocetra(i,j,k,ioxygen)-dano3*1.25
 
 ! Nitrogen fixation followed by remineralisation and nitrification decreases
 ! alkalinity by 1 mole per mole nitrogen fixed (Wolf-Gladrow et al. 2007)
-            ocetra(i,j,k,ialkali)=ocetra(i,j,k,ialkali)-dano3
+                 ocetra(i,j,k,ialkali)=ocetra(i,j,k,ialkali)-dano3
 #ifdef natDIC
-            ocetra(i,j,k,inatalkali)=ocetra(i,j,k,inatalkali)-dano3
+                 ocetra(i,j,k,inatalkali)=ocetra(i,j,k,inatalkali)-dano3
 #endif
 
-            intnfix(i,j) = intnfix(i,j) +                          &
-     &         (ocetra(i,j,k,iano3)-oldocetra)*pddpo(i,j,k)
-
-            ENDIF  
-         ENDIF  
-      ENDDO
-      ENDDO
-!$OMP END PARALLEL DO
-      ENDDO
-
+                 intnfix(i,j) = intnfix(i,j) +                                 &
+                      &         (ocetra(i,j,k,iano3)-oldocetra)*pddpo(i,j,k)
 
+              ENDIF
+           ENDDO
+        ENDIF
+     ENDDO
+  ENDDO
```

Comment:
Maybe not the most important aspect, but I would favor keeping the traditional HAMOCC-style indentation for spatial indices... There is only very few things that were good in HAMOCC regarding code style, but this was one of them. In my opinion

`do k=1,kpke`
`do j=1,kpje`
`do i=1,kpie`

is much preferable over 

`do k=1,kpke`
`___do j=1,kpje`
`______do i=1,kpie`

in terms of readability.

Reply:
I agree with @JorgSchwinger that the HAMOCC part looks ok. Wrt the loops: I guess, we discussed this already a bit... - I was always puzzled with the non-indentation (left me loosing track, where the loops start and end), but I also don't have a really strong opinion about it either and could live with this exception, if needed/wished (as long in other cases indentation is allowed).

Reply:
I re-set the indentation pattern to the traditional HAMOCC-style in cyano.F90. In terms of readability, I think both styles are OK. My code editor auto-indents all loops, and in this case I changed the loop order since `k=1,kmle(i,j)` has to be the inner loop.

Reply:
Thanks Tomas! 

Just to explain a bit more what I mean: In most cases in HAMOCC the i,j-loops (and to a lesser degree also the k-loops) are just loops over spatial indices. The logic is NOT "there is a loop over i where something is done, and in the coarse of doing this there is then also a loop over j", but in almost all cases the logic is "there is an operation done in loops over all horizontal indices". In the first case it would makes sense to indent, but in the second case it is one logical block, where it doesn't really make sense to indent, in my opinion. In principle, these loops could be replaced by modern Fortran ":" notation (not that I think we should do this anytime soon!).

Not that I have a very strong opinion about this, but as mentioned before, this is one of the very few things where I think historical HAMOCC style was good and made sense.

Reply:
Hi @JorgSchwinger , thanks for the detailed explanation and please forgive me (and remind me) when I am unpurposely deviate from that style (due to old habits). I'll try to adjust my developments accordingly.